specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-05-01/managedClusters.json

```diff
@@ -4508,6 +4554,84 @@
         "properties"
       ],
       "description": "The OS option profile."
+    },
+    "EndpointDependency": {
+      "description": "A domain name that AKS agent nodes are reaching at.",
+      "type": "object",
+      "properties": {
+        "domainName": {
+          "description": "The domain name of the dependency.",
+          "type": "string"
+        },
+        "endpointDetails": {
+          "description": "The Ports and Protocols used when connecting to domainName.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EndpointDetail"
+          }
+        }
+      }
+    },
+    "EndpointDetail": {
+      "description": "connect information from the AKS agent nodes to a single endpoint.",
+      "type": "object",
+      "properties": {
+        "ipAddress": {
+          "description": "An IP Address that Domain Name currently resolves to.",
+          "type": "string"
+        },
+        "port": {
+          "format": "int32",
+          "description": "The port an endpoint is connected to.",
+          "type": "integer"
+        },
+        "protocol": {
+          "description": "The protocol used for connection",
+          "type": "string"
+        },
+        "description": {
+          "description": "Description of the detail",
+          "type": "string"
+        }
+      }
+    },
+    "OutboundEnvironmentEndpoint": {
+      "description": "Egress endpoints which AKS agent nodes connect to for common purpose.",
+      "type": "object",
+      "properties": {
+        "category": {
+          "description": "The category of endpoints accessed by the AKS agent node, e.g. azure-resource-management, apiserver, etc.",
+          "type": "string"
+        },
+        "endpoints": {
+          "description": "The endpoints that AKS agent nodes connect to",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EndpointDependency"
+          }
+        }
+      }
+    },
+    "OutboundEnvironmentEndpointCollection": {
+      "description": "Collection of OutboundEnvironmentEndpoint",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
```

Comment:
all entries in this array must follow the resource schema with a top level "id", "name", and "type" property.  They should also be individually GETable by their ID.

Reply:
Hi @pilor ,the only API we have is to get the egress endpoint list outboundNetworkDependenciesEndpoint**s**.  there is no individual resource for outboundNetworkDependenciesEndpoint, no GET outboundNetworkDependenciesEndpoint by ID. We don't expect to have "id", "name", "type" property for it.

Several components in other services are using this schema and we are following the similar.
1. StoragePool: https://github.com/Azure/azure-rest-api-specs/blob/3ae16652e8bea04884a273329d6dead5e9205794/specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2021-04-01-preview/storagepool.json#L993
2. Microsoft.Web: https://github.com/Azure/azure-rest-api-specs/blob/1b0ed8edd58bb7c9ade9a27430759527bd4eec8e/specification/web/resource-manager/Microsoft.Web/stable/2020-10-01/AppServiceEnvironments.json#L1994

Reply:
This API will never be able to transition to a non-readonly API, will not be auditable/enforceable by policy, and can not be ingested into Azure Resource Graph.  Please explicitly ack that you are OK with those restrictions before merging.